base/rational.jl

```diff
@@ -533,3 +533,10 @@ function hash(x::Rational{<:BitInteger64}, h::UInt)
     h = hash_integer(num, h)
     return h
 end
+
+function length(r::AbstractUnitRange{T}) where T<:Rational
```

Comment:
Could use a comment saying that this is only needed for performance.

Reply:
I added a comment in 6a4a9a8.